### DISCUSSION TITLE
Why is mypy checking `pytorch_lightning/plugins/precision/native_amp.py`?

### DISCUSSION THREAD

**Discussion by daniellepintz** [Lightning-AI/pytorch-lightning#9751] [General]
When I run `mypy` (from main directory) I get a bunch of errors, most of which are in the files listed in the "allowlist" here:
https://github.com/PyTorchLightning/pytorch-lightning/blob/60d4cc4f90e75657f08a3cade7e52c91c1a86a34/pyproject.toml#L64-L85

However I also see this file is checked, and I am confused because it is not in the allowlist:
```
pytorch_lightning/plugins/precision/native_amp.py:113: error: Unexpected keyword argument "dtype" for "autocast"  [call-arg]
```

cc @awaelchli

**Comment by awaelchli** [comment]
Hey @daniellepintz 
I don't why it runs differently locally vs the CI. Me and @carmocca  have seen the same. 
mypy 0.910

**Comment by carmocca** [comment]
I'm pretty sure this happens because the `mypy` job does not install any dependencies

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/workflows/code-checks.yml

This means the type stubs for torch, apex, torchmetrics, ... aren't available in CI.

**Comment by daniellepintz** [comment]
Is there any way we can get around this? It is frustrating to see different output locally vs CI

**Reply by carmocca** [reply]
See my post above, the code checks job needs to install dependencies.

Feel free to open a PR if you can take it.

**Reply by daniellepintz** [reply]
I'm a bit confused - doesnt it already install dependencies in https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/workflows/code-checks.yml#L19?

**Reply by carmocca** [reply]
That line just installs `mypy`.

It needs to also install torch, torchvision, torchmetrics, Apex, FairScale, DeepSpeed, ...
